CHANGELOG.md

```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## [4.0.0]
+### Breaking
+- Python 3.8 and earlier are not officially supported, but might still work.
```

Comment:
Since this tool is mostly used in infrastructures that doesn't get updated often by the users, and from the download statistics there seem to be still 10% users on versions earlier than 3.9, I'd suggest we hold on a bit until the usage further drops, then we make a clean cut.

Also something released should either work or not, "might work" doesn't sound right.

Reply:
To be fair, EOL Python users can continue to use junitparser 3.x.

This change does not break the package for pre-3.9, it only drops guarantees. That is meant by the "might".

We can hold changes that definitively break pre-3.9 versions for some more time and still have this change released. This might help to motivate those 10% users to migrate.

Reply:
Ambiguity doesn't sit well with users. The word "might" shouldn't exist in software documentations.

I tend to believe the users are well aware that they're on a dead python version, and they can't update because of certain bigger constraints. Us prodding them to update won't be necessary. 

Also we don't really need to drop the guarantee. Our package is simple and we have decent coverage. If the CI continues to pass with Python 3.8, then 3.8 is supported.

Python 2.7 support was dropped 3 years after EOL. For Python 3.8 it shouldn't be that long, but it could be 1 year or 2, I don't know.